### PR TITLE
Update aiidalab package

### LINKED ISSUE #103: Using the latest packaging version check
Update the `packaging` to `22.0` will cause the issue of parsing the version that `stable` is not a valid version tag. We need take a close look at which app using that and ask the app developer to remove the tag.

### PR DESCRIPTION
**Pull request by danielhollas**
This update includes:
- https://github.com/aiidalab/aiidalab/pull/429
- aiidalab/aiidalab#451 which fixes #103 (tested locally  by running `./build.sh`)

Also includes basic repo cleanup:
- Remove unneeded pre-commit hooks
- trivial CI updates
- update links in README

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,2 @@
-aiidalab==23.3.1
+aiidalab[registry]==24.7.0
 pre-commit==3.6.0
-packaging<22
```
Comment (danielhollas) [inline requirements.txt:3]: This pin is part of `aiidalab` dependencies and so it is not needed here.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -3,20 +3,10 @@ ci:
     autoupdate_schedule: quarterly
 
 repos:
-  - repo: https://github.com/psf/black
```
Comment (danielhollas) [inline .pre-commit-config.yaml:6]: There are no python files in this repo.

**Comment by danielhollas** [comment]
CC @yakutovicha can you please add the deployment group in repo settings? I don't have basic access to this repo (cannot even tag people for review)

**Comment by danielhollas** [comment]
@yakutovicha friendly ping for review. :-)

**Review by yakutovicha — APPROVED** [review]
Sorry, @danielhollas, got lost in all the PRs for review. It is good to go for me 👍

**Pull request merged**